osu.Game/OsuGame.cs

```diff
@@ -346,11 +344,24 @@ public void PresentBeatmap(BeatmapSetInfo beatmap, Predicate<BeatmapInfo> diffic
                     return;
                 }
 
-                // Find first beatmap that matches our predicate.
-                var first = databasedSet.Beatmaps.Find(difficultyCriteria) ?? databasedSet.Beatmaps.First();
+                // Find beatmaps that match our predicate.
+                var beatmaps = databasedSet.Beatmaps.Where(b => difficultyCriteria?.Invoke(b) ?? true);
+                if (!beatmaps.Any())
+                    beatmaps = databasedSet.Beatmaps;
```

Comment:
This should write null to ``difficultyCriteria`` as it wasn't able to provide any difficulties and shouldn't be used again. Would fix the comment below.

Reply:
made it use difficulty criteria only once so this shouldn't be needed.